component/block_cache/block_cache.go

```diff
@@ -1350,9 +1350,10 @@ func (bc *BlockCache) upload(item *workItem) {
 
 	// This block is updated so we need to stage it now
 	err := bc.NextComponent().StageData(internal.StageDataOptions{
-		Name: item.handle.Path,
-		Data: item.block.data[0 : item.block.endIndex-item.block.offset],
-		Id:   item.blockId})
+		Name:   item.handle.Path,
+		Data:   item.block.data[0 : item.block.endIndex-item.block.offset],
+		Offset: uint64(item.block.id),
```

Comment:
Shouldn't this be item.block.offset ?

Reply:
this is what they were using initially so used that. Our encryptor logic consider's this to be block.id.

Reply:
item.block.offset is better, updated. I will update in encryptor too.